internal/suggestions/utils.go

```diff
@@ -0,0 +1,138 @@
+package suggestions
+
+import (
+	"fmt"
+	"github.com/speakeasy-api/speakeasy/internal/utils"
+	"gopkg.in/yaml.v2"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+func getLineNumber(errStr string) (int, error) {
+	lineStr := strings.Split(errStr, "[line ")[1]
+	lineNumStr := strings.Split(lineStr, "]")[0]
+	lineNum, err := strconv.Atoi(lineNumStr)
+	if err != nil {
+		return 0, err
+	}
+	return lineNum, nil
+}
+
+func escapeString(input string) string {
+	re := regexp.MustCompile(`\\([abfnrtv\\'"])`)
+
+	// Replace escape sequences with their unescaped counterparts
+	output := re.ReplaceAllStringFunc(input, func(match string) string {
+		switch match {
+		case `\a`:
+			return "\a"
+		case `\b`:
+			return "\b"
+		case `\f`:
+			return "\f"
+		case `\n`:
+			return "\n"
+		case `\r`:
+			return "\r"
+		case `\t`:
+			return "\t"
+		case `\v`:
+			return "\v"
+		case `\\`:
+			return `\`
+		case `\'`:
+			return `'`
+		case `\"`:
+			return `"`
+		default:
+			// Should never happen, but just in case
+			return match
+		}
+	})
+
+	return output
+}
+
+func formatYaml(input string) (string, error) {
+	// Unmarshal the YAML input into a generic interface{}
+	var data interface{}
+	if err := yaml.Unmarshal([]byte(input), &data); err != nil {
+		return "", err
+	}
+
+	// Marshal the interface{} back to YAML with no extra spaces
+	output, err := yaml.Marshal(data)
+	if err != nil {
+		return "", err
+	}
+	outputStr := strings.TrimSpace(string(output))
+
+	return outputStr, nil
+}
+
+func waitForInput() {
+	var input string
+	for {
+		fmt.Scan(&input)
+		if strings.ToLower(input) == "yes" {
+			// TODO: Update an actual openapi revision, maybe push to speakeasy registry
```

Comment:
could this just modify the document provided via the `-s` argument, so it makes changes on disk.

That would be my expectation anyway as a CLI tool and the way it is normally used by our customers is for doing things locally on their machine

Reply:
It's a little tricky. But it's in progress. Definitely part of the deliverable.

Reply:
It will probably be a follow up PR though